File: Logica_De_Programacao_Java/Desafio_4/src/Desafio_4.java
import java.util.Scanner;

public class Desafio_4 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite o primeiro número: ");
        double num1 = scanner.nextDouble();

        System.out.print("Digite o segundo número: ");
        double num2 = scanner.nextDouble();

        System.out.print("Escolha a operação (+, -, *, /): ");
        char operacao = scanner.next().charAt(0);

        double resultado = 0;
        boolean operacaoValida = true;

        switch (operacao) {
            case '+':
                resultado = num1 + num2;
                break;
            case '-':
                resultado = num1 - num2;
                break;
            case '*':
                resultado = num1 * num2;
                break;
            case '/':
                if (num2 == 0) {
                    System.out.println("Erro: Divisão por zero não permitida!");
                    operacaoValida = false;
                } else {
                    resultado = num1 / num2;
                }
                break;
            default:
                System.out.println("Erro: Operação inválida!");
                operacaoValida = false;
                break;
        }

        if (operacaoValida) {
            System.out.println("Resultado: " + resultado);
        }

        scanner.close();
    }
}

File: Logica_de_Programacao_Python/Desafio_4/desafio_4/Desafio4.py
def adicao(a, b):
    return a + b

def subtracao(a, b):
    return a - b

def multiplicacao(a, b):
    return a * b

def divisao(a, b):
    if b != 0:
        return a / b
    else:
        return "Erro: Divisão por zero não é permitida."

def main():
    print('-------------------------------')
    print("Bem-vindo à calculadora!")
    print('-------------------------------\n')

    print("Selecione a operação que deseja realizar:")
    print("1 - Adição (+)")
    print("2 - Subtração (-)")
    print("3 - Multiplicação (*)")
    print("4 - Divisão (/)\n")

    operacao = input("Digite o número da operação escolhida: ")

    try:
        num1 = float(input("\nDigite o primeiro número: "))
        num2 = float(input("Digite o segundo número: "))

        match operacao:
            case "1":
                resultado = adicao(num1, num2)
                simbolo = "+"
            case "2":
                resultado = subtracao(num1, num2)
                simbolo = "-"
            case "3":
                resultado = multiplicacao(num1, num2)
                simbolo = "*"
            case "4":
                resultado = divisao(num1, num2)
                simbolo = "/"
            case _:
                print("\nOperação inválida. Tente novamente.")
                return

        # Mostra o resultado
        print(f"\nResultado: {num1} {simbolo} {num2} = {resultado}")

    except ValueError:
        print("\nErro: Por favor, insira números válidos.")

    print("\nObrigado por usar a calculadora!")

if __name__ == "__main__":
    main()